### PR TITLE
Optimize cocoapods logic in flutter doctor.

### LINKED ISSUE #23462: flutter doctor reports `NoSuchMethodError: The method '<' was called on null` when checking cocoapods
## Steps to Reproduce

First run 'flutter doctor', I got this error

## Logs
```
 flutter doctor -v

[✓] Flutter (Channel beta, v0.9.4, on Mac OS X 10.14 18A391, locale zh-Hans-CN)
    • Flutter version 0.9.4 at /Users/xxx/dev/flutter
    • Framework revision f37c235c32 (4 weeks ago), 2018-09-25 17:45:40 -0400
    • Engine revision 74625aed32
    • Dart version 2.1.0-dev.5.0.flutter-a2eb050044

[✓] Android toolchain - develop for Android devices (Android SDK 28.0.3)
    • Android SDK at /Users/xxx/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)
    • All Android licenses accepted.


Oops; flutter has exited unexpectedly.
Sending crash report to Google.
Crash report sent (report ID: c21f887debac35e0)
⣾Unhandled exception:
NoSuchMethodError: The method '<' was called on null.
Receiver: null
Tried calling: <(Instance of 'Version')
#0      Object.noSuchMethod (dart:core/runtime/libobject_patch.dart:50:5)
#1      CocoaPods.evaluateCocoaPodsInstallation (package:flutter_tools/src/ios/cocoapods.dart:69:28)
<asynchronous suspension>
#2      CocoaPodsValidator.validate (package:flutter_tools/src/ios/ios_workflow.dart:210:12)
<asynchronous suspension>
#3      GroupedValidator.validate (package:flutter_tools/src/doctor.dart:265:52)
#4      _AsyncAwaitCompleter.start (dart:async/runtime/libasync_patch.dart:49:6)
#5      GroupedValidator.validate (package:flutter_tools/src/doctor.dart:262:36)
#6      Doctor.startValidatorTasks (package:flutter_tools/src/doctor.dart:105:52)
#7      Doctor.diagnose (package:flutter_tools/src/doctor.dart:162:41)
#8      _AsyncAwaitCompleter.start (dart:async/runtime/libasync_patch.dart:49:6)
#9      Doctor.diagnose (package:flutter_tools/src/doctor.dart:152:24)
#10     _doctorText.<anonymous closure> (package:flutter_tools/runner.dart:187:26)
#11     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:142:29)
#12     _AsyncAwaitCompleter.start (dart:async/runtime/libasync_patch.dart:49:6)
#13     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:142:7)
#14     _rootRun (dart:async/zone.dart:1124:13)
#15     _CustomZone.run (dart:async/zone.dart:1021:19)
#16     _runZoned (dart:async/zone.dart:1516:10)
#17     runZoned (dart:async/zone.dart:1463:12)
#18     AppContext.run (package:flutter_tools/src/base/context.dart:141:18)
#19     _AsyncAwaitCompleter.start (dart:async/runtime/libasync_patch.dart:49:6)
#20     AppContext.run (package:flutter_tools/src/base/context.dart:129:19)
#21     _doctorText (package:flutter_tools/runner.dart:186:19)
#22     _AsyncAwaitCompleter.start (dart:async/runtime/libasync_patch.dart:49:6)
#23     _doctorText (package:flutter_tools/runner.dart:182:27)
#24     _createLocalCrashReport (package:flutter_tools/runner.dart:164:32)
#25     _AsyncAwaitCompleter.start (dart:async/runtime/libasync_patch.dart:49:6)
#26     _createLocalCrashReport (package:flutter_tools/runner.dart:149:37)
#27     _handleToolError (package:flutter_tools/runner.dart:119:33)
#28     _asyncThenWrapperHelper.<anonymous closure> (dart:async/runtime/libasync_patch.dart:77:64)
#29     _rootRunUnary (dart:async/zone.dart:1132:38)
#30     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#31     _FutureListener.handleValue (dart:async/future_impl.dart:129:18)
#32     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:642:45)
#33     Future._propagateToListeners (dart:async/future_impl.dart:671:32)
#34     Future._complete (dart:async/future_impl.dart:476:7)
#35     _SyncCompleter.complete (dart:async/future_impl.dart:51:12)
#36     _AsyncAwaitCompleter.complete (dart:async/runtime/libasync_patch.dart:28:18)
#37     _completeOnAsyncReturn (dart:async/runtime/libasync_patch.dart:295:13)
#38     CrashReportSender.sendReport (package:flutter_tools/src/crash_reporting.dart)
#39     _asyncThenWrapperHelper.<anonymous closure> (dart:async/runtime/libasync_patch.dart:77:64)
#40     _rootRunUnary (dart:async/zone.dart:1132:38)
#41     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#42     _FutureListener.handleValue (dart:async/future_impl.dart:129:18)
#43     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:642:45)
#44     Future._propagateToListeners (dart:async/future_impl.dart:671:32)
#45     Future._complete (dart:async/future_impl.dart:476:7)
#46     Stream.fold.<anonymous closure> (dart:async/stream.dart:726:18)
#47     _rootRun (dart:async/zone.dart:1120:38)
#48     _CustomZone.run (dart:async/zone.dart:1021:19)
#49     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#50     _BufferingStreamSubscription._sendDone.sendDone (dart:async/stream_impl.dart:389:13)
#51     _BufferingStreamSubscription._sendDone (dart:async/stream_impl.dart:399:15)
#52     _BufferingStreamSubscription._close (dart:async/stream_impl.dart:283:7)
#53     _SinkTransformerStreamSubscription._close (dart:async/stream_transformers.dart:96:11)
#54     _EventSinkWrapper.close (dart:async/stream_transformers.dart:23:11)
#55     _StringAdapterSink.close (dart:convert/string_conversion.dart:281:11)
#56     _Utf8ConversionSink.close (dart:convert/string_conversion.dart:336:20)
#57     _ConverterStreamEventSink.close (dart:convert/chunked_conversion.dart:94:18)
#58     _SinkTransformerStreamSubscription._handleDone (dart:async/stream_transformers.dart:141:24)
#59     _rootRun (dart:async/zone.dart:1120:38)
#60     _CustomZone.run (dart:async/zone.dart:1021:19)
#61     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#62     _BufferingStreamSubscription._sendDone.sendDone (dart:async/stream_impl.dart:389:13)
#63     _BufferingStreamSubscription._sendDone (dart:async/stream_impl.dart:399:15)
#64     _BufferingStreamSubscription._close (dart:async/stream_impl.dart:283:7)
#65     _ForwardingStream._handleDone (dart:async/stream_pipe.dart:106:10)
#66     _ForwardingStreamSubscription._handleDone (dart:async/stream_pipe.dart:172:13)
#67     _rootRun (dart:async/zone.dart:1120:38)
#68     _CustomZone.run (dart:async/zone.dart:1021:19)
#69     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#70     _BufferingStreamSubscription._sendDone.sendDone (dart:async/stream_impl.dart:389:13)
#71     _BufferingStreamSubscription._sendDone (dart:async/stream_impl.dart:399:15)
#72     _BufferingStreamSubscription._close (dart:async/stream_impl.dart:283:7)
#73     _SinkTransformerStreamSubscription._close (dart:async/stream_transformers.dart:96:11)
#74     _EventSinkWrapper.close (dart:async/stream_transformers.dart:23:11)
#75     _ByteAdapterSink.close (dart:convert/byte_conversion.dart:70:11)
#76     _FilterSink.close (dart:io/data_transformer.dart:636:11)
#77     _ConverterStreamEventSink.close (dart:convert/chunked_conversion.dart:94:18)
#78     _SinkTransformerStreamSubscription._handleDone (dart:async/stream_transformers.dart:141:24)
#79     _rootRun (dart:async/zone.dart:1120:38)
#80     _CustomZone.run (dart:async/zone.dart:1021:19)
#81     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#82     _BufferingStreamSubscription._sendDone.sendDone (dart:async/stream_impl.dart:389:13)
#83     _BufferingStreamSubscription._sendDone (dart:async/stream_impl.dart:399:15)
#84     _BufferingStreamSubscription._close (dart:async/stream_impl.dart:283:7)
#85     _ForwardingStream._handleDone (dart:async/stream_pipe.dart:106:10)
#86     _ForwardingStreamSubscription._handleDone (dart:async/stream_pipe.dart:172:13)
#87     _rootRun (dart:async/zone.dart:1120:38)
#88     _CustomZone.run (dart:async/zone.dart:1021:19)
#89     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#90     _BufferingStreamSubscription._sendDone.sendDone (dart:async/stream_impl.dart:389:13)
#91     _BufferingStreamSubscription._sendDone (dart:async/stream_impl.dart:399:15)
#92     _BufferingStreamSubscription._close (dart:async/stream_impl.dart:283:7)
#93     _SyncStreamController._sendDone (dart:async/stream_controller.dart:771:19)
#94     _StreamController._closeUnchecked (dart:async/stream_controller.dart:628:7)
#95     _StreamController.close (dart:async/stream_controller.dart:621:5)
#96     _HttpParser._closeIncoming (dart:_http/http_parser.dart:1037:23)
#97     _HttpParser._doParse (dart:_http/http_parser.dart:752:11)
#98     _HttpParser._parse (dart:_http/http_parser.dart:318:7)
#99     _HttpParser._onData (dart:_http/http_parser.dart:810:5)
#100    _rootRunUnary (dart:async/zone.dart:1132:38)
#101    _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#102    _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)
#103    _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)
#104    _BufferingStreamSubscription._add (dart:async/stream_impl.dart:263:7)
#105    _SyncStreamController._sendData (dart:async/stream_controller.dart:763:19)
#106    _StreamController._add (dart:async/stream_controller.dart:639:7)
#107    _StreamController.add (dart:async/stream_controller.dart:585:5)
#108    _Socket._onData (dart:io/runtime/binsocket_patch.dart:1714:41)
#109    _rootRunUnary (dart:async/zone.dart:1132:38)
#110    _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#111    _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)
#112    _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)
#113    _BufferingStreamSubscription._add (dart:async/stream_impl.dart:263:7)
#114    _SyncStreamController._sendData (dart:async/stream_controller.dart:763:19)
#115    _StreamController._add (dart:async/stream_controller.dart:639:7)
#116    _StreamController.add (dart:async/stream_controller.dart:585:5)
#117    _RawSecureSocket._sendReadEvent (dart:io/secure_socket.dart:1004:19)
#118    _rootRun (dart:async/zone.dart:1120:38)
#119    _CustomZone.run (dart:async/zone.dart:1021:19)
#120    _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#121    _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#122    _rootRun (dart:async/zone.dart:1124:13)
#123    _CustomZone.run (dart:async/zone.dart:1021:19)
#124    _CustomZone.bindCallback.<anonymous closure> (dart:async/zone.dart:947:23)
#125    Timer._createTimer.<anonymous closure> (dart:async/runtime/libtimer_patch.dart:21:15)
#126    _Timer._runTimers (dart:isolate/runtime/libtimer_impl.dart:382:19)
#127    _Timer._handleMessage (dart:isolate/runtime/libtimer_impl.dart:416:5)
#128    _RawReceivePortImpl._handleMessage (dart:isolate/runtime/libisolate_patch.dart:171:12)
```

### PR DESCRIPTION
**Pull request by kangwang1988**
Optimize cocoapods logic in flutter doctor.
Fixes #23462 

In fact, I've also experience same problems. In my company, developers are using customized cocoapods, when running `pod --version`, it might output "xxx plugin loaded.\n1.5.0", which will cause failure in `flutter doctor`,etc. In fact, when you run `pod install` directly, it works fine.

### CODE REVIEW

File: packages/flutter_tools/lib/src/base/user_messages.dart [hunk 1]
```diff
@@ -170,6 +170,11 @@ class UserMessages {
       '$consequence\n'
       'To install:\n'
       '$installInstructions';
+  String cocoaPodsUnknownVersion(String consequence, String upgradeInstructions) =>
+      'CocoaPods unknown version installed.\n'
```
Comment (xster) [inline packages/flutter_tools/lib/src/base/user_messages.dart]: `Unknown CocoaPods version installed.\n`

File: packages/flutter_tools/lib/src/ios/cocoapods.dart [hunk 2]
```diff
@@ -24,6 +24,9 @@ const String noCocoaPodsConsequence = '''
   Without resolving iOS dependencies with CocoaPods, plugins will not work on iOS.
   For more info, see https://flutter.io/platform-plugins''';
 
+const String unknownCocoaPodsConsequence = '''
+  Make sure that `pod --version` always return pure version like :''';
```
Comment (xster) [inline packages/flutter_tools/lib/src/ios/cocoapods.dart]: Sounds a bit too imperative maybe? I assume everything will still work and we don't need to force the user to make their `pod --version` output `1.5.3` right? It's just to remove the warning? 

Maybe 
```
Flutter is unable to determined the installed CocoaPods's version. 
Ensure that the output of 'pod --version' contains only digits and . to be recognized by Flutter.
```

File: packages/flutter_tools/lib/src/ios/cocoapods.dart [hunk 3]
```diff
@@ -112,6 +119,15 @@ class CocoaPods {
           emphasis: true,
         );
         return false;
+      case CocoaPodsStatus.unknownVersion:
+        printError(
+          'Warning: CocoaPods unknown version installed.\n'
```
Comment (xster) [inline packages/flutter_tools/lib/src/ios/cocoapods.dart]: `Warning: unknown CocoaPods version installed.\n'

**Review by xster — APPROVED** [review]
Thanks

**Comment by kangwang1988** [comment]
@xster 
All issues resolved.

**Comment by xster** [comment]
Thanks. The tree break seems like a flake. Merging

**Pull request merged**